tests/checkpoint/conftest.py

```diff
@@ -135,3 +138,84 @@ def titanic_spark_data_context_with_v013_datasource_with_checkpoints_v1_with_emp
     # noinspection PyProtectedMember
     context._save_project_config()
     return context
+
+
+@pytest.fixture
+def context_with_single_taxi_csv_spark(
+    empty_data_context, tmp_path_factory, spark_session
+):
+    context = empty_data_context
+
+    base_directory = str(tmp_path_factory.mktemp("test_checkpoint_spark"))
```

Comment:
Do we have an in memory file system we can use?

Reply:
I believe so, but we currently don't use it extensively... I can look into it as an option for this PR

Reply:
Looked into it and we are using `pyarrow` to do some in-memory filesystem tests, mostly for S3 and DBFS.  Using it for Filesystem (like in this case) would be a bit of a departure from existing patterns, but one that I think there would be. Would it be ok to create a follow-up task to re-visit the possibility of using an in-memory filesystem for these types of tests? (not just this fixture, but for other Filesystem tests)

Reply:
Yes, a follow up is fine.